### ISSUE TITLE
Skeleton is broken: no way to tell what a class inherits from its parent

### ISSUE THREAD

**Issue by llorllale** [yegor256/jpeek#187]
We currently can't tell whether a field or method defined in a superclass and being accessed from a subclass method is actually inherited.

This is relevant for some metrics, eg. `TCC`. On the topic of **Inheritance and Cohesion**, the `TCC` metric says:

>A subclass inherits methods and instance variables from its superclass(es). We have several options for evaluating cohesion of a subclass. We can (1) include all inherited components in the subclass in our evaluation, (2) include only methods and instance variables defined in the subclass, or (3) include inherited instance variables but not inherited methods. The class cohesion measures that we develop can be applied using any one of these options.

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by yegor256** [comment]
@llorllale you are saying that we need some markers for attributes and methods, that will tell us which of them are inherited from a super class. Correct?

**Comment by llorllale** [comment]
@yegor256 I think we can have either, a) xsd:attribute on class indicating its superclass, or b) xsd:boolean on attributes+methods indicating if they're inherited

Any of those would work I guess.

**Comment by yegor256** [comment]
@llorllale I think we need four things

  * `class/parent`
  * `class/interfaces/interface`
  * `class/methods/method/@inherited`
  * `class/attributes/attribute/@inherited`

Agree?

**Comment by llorllale** [comment]
@yegor256 for the purposes I stated in the description, I would just need `method/@inherited` and `attribute/@inherited` (assuming they're `xsd:boolean`).

But if you are planning on adding all those, my question is: how would I know from *what* I inherited the method or attribute from? The parent or one of the interfaces?

**Comment by yegor256** [comment]
@llorllale we don't need to know that, right? Just boolean attributes will be enough?

**Comment by llorllale** [comment]
@yegor256 correct. For now, just boolean attributes on `<method>` and `<attribute>` will be enough

**Comment by 0crat** [comment]
Job [#187](https://github.com/yegor256/jpeek/issues/187) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://datum.zerocracy.com/pages/policy.html#29): +15 points just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale), total is [+620](https://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
The job [#187](https://github.com/yegor256/jpeek/issues/187) assigned to @mesut[/z](https://www.0crat.com/u/mesut), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/1871149229); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
@mesut[/z](https://www.0crat.com/u/mesut) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
> @0crat wait ([here](https://github.com/yegor256/jpeek/issues/187#issuecomment-405206028))

@mesut The impediment for [#187](https://github.com/yegor256/jpeek/issues/187) was registered successfully by @mesut[/z](https://www.0crat.com/u/mesut)

**Comment by 0crat** [comment]
The architect of the project has changed; @yegor256[/z](https://www.0crat.com/u/yegor256) is not at this role anymore; @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) is the architect now
